### PR TITLE
I18n support

### PR DESCRIPTION
**Pull request by Moxide**
In reference to #136, a  first attempt to bring i18n support, using Gettext.
Issues :
- No error messages translation (don't need those, in my case)
- No javascript strings translation (I don't know how to do this, perhaps generate the js files from Elixir so we could leverage Gettext ?)
- This a WIP, there could be some errors, some untranslated strings left...etc. ;-)

The .pot file is in priv/gettext, a french .po file is provided.
To test, add the following your app config.exs :
```
config :ex_admin, ExAdmin.Gettext,
  default_locale: "fr_FR"
```

### CODE REVIEW

File: lib/ex_admin/themes/admin_lte2/index.ex [hunk 1]
```diff
@@ -24,11 +25,11 @@ defmodule ExAdmin.Theme.AdminLte2.Index do
     div ".blank_slate_container" do
       span ".blank_slate" do
         unless is_nil(conn.params["q"]) and is_nil(conn.params["scope"]) do
-          text "No #{humanize label} found."
+          text (gettext"No %{label} found.", label: (humanize label))
```
Comment (smpallen99) [inline lib/ex_admin/themes/admin_lte2/index.ex]: Why would you not localize the labels here and below?

**Comment by smpallen99** [comment]
@Moxide Do you know of any reason why there isn't a sigil for gettext? Wouldn't it read read much nicer with something like `~g(this is localized text)`?

**Comment by smpallen99** [comment]
@Moxide Just tried with one of my projects. Added the config above. No french.

**Comment by Moxide** [comment]
Just saw your comments.

> Wouldn't it read read much nicer with something like ~g(this is localized text) ?

Yes it would ;-) Unfortunately it seems there isn't a shorter keyword in Elixir's Gettext implementation.

**Comment by tmbb** [comment]
> No javascript strings translation (I don't know how to do this, perhaps generate the js files from Elixir so we could leverage Gettext ?)

In my project, I generate a hidden `div`s in the HTML template (the .eex file) containing the UI strings wrapped in `gettext`. The, in the JS scripts get the strings from the DOM at runtime. The JS files remain static this way. This works well for a custom webpage, but for I don't know how it would work for something like ex_admin.

**Comment by smpallen99** [comment]
I do a similar trick, only I create a JS hash with all the messages and render it in the layout. I know there are are some node models, but I have not tried them, nor I know how easy it would be to integrate with the phoenix stuff.

**Comment by tmbb** [comment]
@smpallen99 This will break backward compatibility, but shouldn't we be using `dgettext("ex_admin", ...)` instead of `gettext(...)`?

**Comment by smpallen99** [comment]
@tmbb Yes we should be using `dgettext` instead of `gettext`.

**Comment by smpallen99** [comment]
@Moxide Just noticed your comment about a shorter `gettext`. I solved this problem and am using it in one of my closed-source projects at work. I created a `~g` sigil macro that allows me to use `~g"Some text"`. However, I don't have support for `ngettext` or `dgettext`.

**Comment by tmbb** [comment]
the `~g` sigil seems great for an application (which can probably get by using `gettext` alone just fine), but for a library using `dgettext` I think the brevity of something like `~g` is not that useful, because you're always going to have the additional clutter of the domain name ("ex_admin" in this case)...

**Comment by tmbb** [comment]
People interested in internationalizing ex_admin might be interested in this discussion, at the coherence repo: https://github.com/smpallen99/coherence/issues/185

**Pull request merged**